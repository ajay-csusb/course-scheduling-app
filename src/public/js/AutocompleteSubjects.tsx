import { IItemRendererProps } from '@blueprintjs/select';
import React from 'react';
import { MenuItem } from '@blueprintjs/core';
import { ISubject } from './Subject';

export function filterSubjects(query: string, subject: ISubject): boolean {
  const parsedQuery = query.toLowerCase().trim();
  if (subject.name.toLowerCase().trim().indexOf(parsedQuery) >= 0) {
    return true;
  }
  if (subject.abbr.toLowerCase().trim().indexOf(parsedQuery) >= 0) {
    return true;
  }
  return false;
}

function renderSubjects(subject: ISubject, classRenderer: IItemRendererProps): JSX.Element {
  return (
    <MenuItem
      active={false}
      disabled={false}
      label={subject.abbr}
      key={subject.abbr}
      onClick={classRenderer.handleClick}
      text={subject.name}
      className="search-autocomplete-items"
    />
  );
}

export function renderSubject(subject: ISubject): string {
  return `${subject.name.trim()} (${subject.abbr.trim()})`;
}

export const autocompleteSubjectsProps = {
  itemRenderer: renderSubjects,
  itemPredicate: filterSubjects,
  inputValueRenderer: renderSubject,
};
