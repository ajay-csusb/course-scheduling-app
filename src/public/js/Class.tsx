import * as ClassSearchUtils from './ClassSearchUtils';
export interface IClass {
  quarter: string;
  subject: string;
  catalogNo: string;
  classSection: string;
  textbook: string;
  employeeId: string;
  profile?: string;
  classStartTime: string;
  classEndTime: string;
  courseOfferNo?: number;
  classNumber?: number;
  enrolledCapacity: number;
  enrolledTotal: number;
  waitlistCapacity: number;
  waitlistTotal: number;
  classMeetingNo: number;
  csuUnits: number;
  amount: number;
  courseId: string;
  sessionCode: string;
  description: string;
  ssrComponent: string;
  classType?: string;
  classStat?: string;
  campus: string;
  longDescription?: string;
  schedulePrint?: string;
  academicGroup?: string;
  academicOrg?: string;
  instructionMode: string;
  sbCourseZccm?: string;
  consent?: string;
  facilityId: string;
  buildingCode?: string;
  room?: string;
  mon: string;
  tues: string;
  wed: string;
  thurs: string;
  fri: string;
  sat: string;
  sun: string;
  instructorName: string;
  instructorEmployeeId?: string;
  instructorAltName?: string;
}

export interface IMeetingDate {
  all: boolean;
  mon: boolean;
  tue: boolean;
  wed: boolean;
  thu: boolean;
  fri: boolean;
  sat: boolean;
  sun: boolean;
}

export class Class {
  static classesUrl = 'http://webdx.csusb.edu/ClassSchedule/getCurrentCS';

  private classInfo: IClass;

  constructor(classData: IClass) {
    this.classInfo = classData;
  }

  static transformToClass(object: any): IClass {
    const result: IClass = {
      quarter : object.strm,
      subject : object.subject,
      catalogNo : object.catalog_NBR,
      classSection : object.class_SECTION,
      textbook : object.book,
      employeeId : object.emplid,
      profile : object.profile_PATH,
      classStartTime : object.class_START_TIME,
      classEndTime : object.class_END_TIME,
      courseOfferNo : object.crse_OFFER_NBR,
      classNumber : object.class_NBR,
      enrolledCapacity : object.enrl_CAP,
      enrolledTotal : object.enrl_TOT,
      waitlistCapacity : object.wait_CAP,
      waitlistTotal : object.wait_TOT,
      classMeetingNo : object.class_MTG_NBR,
      csuUnits : object.csu_APDB_CMP_UNITS,
      amount : object.flat_AMT,
      courseId : object.crse_ID,
      sessionCode : object.session_CODE,
      description : object.descr,
      ssrComponent : object.ssr_COMPONENT,
      classType : object.class_TYPE,
      classStat : object.class_STAT,
      campus : object.campus,
      longDescription : object.descrlong_VC,
      schedulePrint : object.schedule_PRINT,
      academicGroup : object.acad_GROUP,
      academicOrg : object.acad_ORG,
      instructionMode : object.instruction_MODE,
      sbCourseZccm : object.sb_CRSE_ZCCM,
      consent : object.consent,
      facilityId : object.facility_ID,
      buildingCode : object.bldg_CD,
      room : object.room,
      mon : object.mon,
      tues : object.tues,
      wed : object.wed,
      thurs : object.thurs,
      fri : object.fri,
      sat : object.sat,
      sun : object.sun,
      instructorName : object.name,
      instructorEmployeeId : object.emplid2,
      instructorAltName : object.name2,
    };
    return result;
  }

  static getAllClasses(onSuccess: (response: any) => void,
                       onFailure: (error: string) => void, url: string = this.classesUrl): void {
    const currentQuarterId = localStorage.getItem('currentQuarterId') ;
    const params = {
      currentPage: '1',
      maximunPerPage: '4000',
      conditions: ['strm = ' + currentQuarterId],
    };
    ClassSearchUtils.fetchWithArg(url, params, onSuccess, onFailure);
  }

  public getClassMeetingDates(): string {
    let days = '';
    let day = '';
    if (this.classInfo.mon === 'Y') {
      day += ' M';
    }
    if (this.classInfo.tues === 'Y') {
      day += ' Tu';
    }
    if (this.classInfo.wed === 'Y') {
      day += ' W';
    }
    if (this.classInfo.thurs === 'Y') {
      day += ' Th';
    }
    if (this.classInfo.fri === 'Y') {
      day += ' F';
    }
    if (this.classInfo.sat === 'Y') {
      day += ' Sa';
    }
    if (this.classInfo.sun === 'Y') {
      day += ' Su';
    }
    // Strip leading and trailing spaces, and replace spaces in between with -
    days = day.replace(/(^\s+)/g, '').replace(/(\s+)/g, '-');
    if (day === '') {
      days = 'N/A';
    }
    return days;
  }
  public getClassMeetingTimes(): string {
    let meetingTimes = 'N/A';
    if (!this.areStartTimeEndTimeEmpty()) {
      meetingTimes = `${this.classInfo.classStartTime} - ${this.classInfo.classEndTime}`.toLowerCase();
    }
    return meetingTimes;
  }

  private areStartTimeEndTimeEmpty(): boolean {
    return (this.classInfo.classStartTime.length === 0 && this.classInfo.classEndTime.length === 0);
  }
}
