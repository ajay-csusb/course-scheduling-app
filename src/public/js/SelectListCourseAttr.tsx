import React from 'react';
import { Label, HTMLSelect, IOptionProps } from '@blueprintjs/core';

export interface ISelectListCourseAttrProps {
  courseAttr: string;
  onChangeOfCourseAttr: (event: any) => void;
}

export class SelectListCourseAttr extends React.Component<ISelectListCourseAttrProps, {}> {

  private readonly courseAttr: IOptionProps[] = [
      { label: 'All', value: 'all' },
      { label: 'Early Start Program', value: 'ESP' },
      { label: 'Fully Online', value: 'FONL' },
      { label: 'General Education', value: 'GE' },
      { label: 'Graduate classes', value: 'PBAC' },
      { label: 'Open University Course', value: 'EXED'},
      { label: 'Service Learning', value: 'CSLI' },
      { label: 'Study Abroad', value: 'SA' },
      { label: 'Zero Cost Course Materials', value: 'ZCCM' },
      { label: 'eBook', value: 'EBK' },
      { label: 'Undergraduate classes', value: 'UGRD' },
    ];

  public render(): React.ReactNode {
    return(
      <Label>
        Course Attribute
        <HTMLSelect
          value={this.props.courseAttr}
          className="course-attribute"
          onChange={this.props.onChangeOfCourseAttr}
          options={this.courseAttr}
        />
      </Label>
    );
  }

}
