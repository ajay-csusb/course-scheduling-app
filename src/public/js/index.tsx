import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { ClassSearchContainer } from './ClassSearchContainer';
import { Utils } from './Utils';

Utils.loadExternalTools();
if (Utils.isProd()) {
  const ErrorBoundary = Utils.loadBugsnagComponent();
  ReactDOM.render(<ErrorBoundary><ClassSearchContainer /></ErrorBoundary>,
    document.querySelector('#class-search-form-wrapper'));
} else {
  ReactDOM.render(<ClassSearchContainer />, document.querySelector('#class-search-form-wrapper'));
}
