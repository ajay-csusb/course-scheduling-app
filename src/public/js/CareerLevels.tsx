import React from 'react';
import { ControlGroup, Checkbox, Alignment } from '@blueprintjs/core';
import { ICareerLevels } from './Class';

export interface ICareerLevelsProps {
  careerLevelsOptions: ICareerLevels;
  onChangeOfCareerLevelsOptions: (event: any) => void;
  onKeyDown: (event: React.KeyboardEvent) => void;
}

export class CareerLevels extends React.Component<ICareerLevelsProps, {}> {
  constructor(props: ICareerLevelsProps) {
    super(props);
    this.handleChangeOfCareerLevelsOptions = this.handleChangeOfCareerLevelsOptions.bind(this);
  }

  public render(): React.ReactNode {
    return (
      <ControlGroup fill={true} vertical={false} className="career-levels-filter-wrapper">
        <div className="label">Career Levels</div>
        <div id="career-levels-filter">
          <div className="form-item">
            <Checkbox
              id="ugrd"
              value="ugrd"
              className="career-levels-checkboxes"
              onChange={(event: React.FormEvent) => this.handleChangeOfCareerLevelsOptions(event)}
              checked={this.props.careerLevelsOptions.ugrd}
              onKeyDown={this.props.onKeyDown}
              inline={true}
              alignIndicator={Alignment.LEFT}
              label="Undergraduate"
            />
          </div>
          <div className="form-item">
            <Checkbox
              id="pbac"
              value="pbac"
              className="career-levels-checkboxes"
              onChange={(event: React.FormEvent) => this.handleChangeOfCareerLevelsOptions(event)}
              checked={this.props.careerLevelsOptions.pbac}
              onKeyDown={this.props.onKeyDown}
              inline={true}
              alignIndicator={Alignment.LEFT}
              label="Post baccalaureate"
            />
          </div>
          <div className="form-item">
            <Checkbox
              id="exed"
              value="exed"
              className="career-levels-checkboxes"
              onChange={(event: React.FormEvent) => this.handleChangeOfCareerLevelsOptions(event)}
              checked={this.props.careerLevelsOptions.exed}
              onKeyDown={this.props.onKeyDown}
              inline={true}
              alignIndicator={Alignment.LEFT}
              label="Extended education"
            />
          </div>
        </div>
      </ControlGroup>
    );
  }

  private handleChangeOfCareerLevelsOptions(event: React.FormEvent): void {
    this.props.onChangeOfCareerLevelsOptions(event);
  }
}
