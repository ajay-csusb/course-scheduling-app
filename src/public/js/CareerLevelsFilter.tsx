import { IClass, ICareerLevels } from './Class';

export function filter(classes: IClass[], careerLevelsOptions: ICareerLevels): IClass[] {
  const result: IClass[] = [];

  if (!atLeastOneOptionSelected(careerLevelsOptions)) {
    return classes;
  }

  classes.forEach(_class => {
    if (careerLevelsOptions.ugrd && _class.degreeType.toLowerCase() === 'ugrd') {
      result.push(_class);
    }
    if (careerLevelsOptions.pbac && _class.degreeType.toLowerCase() === 'pbac') {
      result.push(_class);
    }
    if (careerLevelsOptions.exed && _class.degreeType.toLowerCase() === 'exed') {
      result.push(_class);
    }
  });

  return result;
}

function atLeastOneOptionSelected(careerLevelsOptions: ICareerLevels) {
  return careerLevelsOptions.ugrd || careerLevelsOptions.pbac || careerLevelsOptions.exed;
}
