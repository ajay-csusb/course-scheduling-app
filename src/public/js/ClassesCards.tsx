import * as React from 'react';
import * as ClassSearchUtils from './ClassSearchUtils';
import { IClass, Class } from './Class';
import { Popover, Classes } from '@blueprintjs/core';
import { Utils } from './Utils';

interface IClassesCardsProps {
  classes: IClass;
  currentTerm: string;
}

export class ClassesCards extends React.Component<IClassesCardsProps> {
  readonly searchURL = 'https://search.csusb.edu';
  private classDetails: IClass;

  public constructor(props: IClassesCardsProps) {
    super(props);
    this.classDetails = this.props.classes;
  }

  public render(): JSX.Element {
    const courseHeaderMarkup = this.getCourseHeaderMarkup();
    const courseBodyMarkup = this.getCourseBodyMarkup();
    const courseInfoMarkup = this.getCourseInfoMarkup();
    return (
      <div className="course result-item">
        <div className="item-header">
          {courseHeaderMarkup}
          {courseInfoMarkup}
        </div>
        <div className="item-body">
          {courseBodyMarkup}
        </div>
      </div>
    );
  }
  public getCourseBodyMarkup(): JSX.Element {
    const classObj = new Class(this.classDetails);
    const campus = ClassSearchUtils.getCampusName(this.classDetails.campus);
    const days = classObj.getClassMeetingDates();
    const time = classObj.getClassMeetingTimes();
    const roomNumberMarkup = this.getRoomNumberMarkup();
    const instructionMode = ClassSearchUtils.getInstructionMode(this.classDetails);
    const textBookMarkup = this.getTextBookMarkup();
    const instructorMarkup = this.getInstructorMarkup();
    const sessionMarkup = this.getSessionMarkup();
    let courseAttrMarkup = null;
    if (this.classDetails.courseAttr.length !== 0) {
      courseAttrMarkup = <li><span>Course Attribute </span>{this.classDetails.courseAttr}</li>;
    }
    const courseMaterialIconMarkup = this.getCourseMaterialIconMarkup();
    return (
      <React.Fragment>
        <ul className="course-desc">
          <li><span>Units </span>{this.classDetails.csuUnits}</li>
          <li><span>Meeting Time </span>{time}</li>
          <li><span>Meeting Days </span>{days}</li>
          {roomNumberMarkup}
          <li><span>Campus </span>{campus}</li>
          <li>{instructorMarkup}</li>
          <li><span>Instruction Mode </span>{instructionMode}</li>
          {courseAttrMarkup}
          {sessionMarkup}
        </ul>
        <div className="course-books">
          {textBookMarkup}
          {courseMaterialIconMarkup}
        </div>

      </React.Fragment>
    );
  }

  public getCourseInfoMarkup(): JSX.Element {
    const classStatus = this.getClassStatusMarkup();
    const classAvailability = this.getClassAvailabilityMarkup();
    return (
      <div className="course-info">
        {classStatus}
        {classAvailability}
      </div>
    );
  }

  public getTextBookMarkup(): JSX.Element {
    return (
      <React.Fragment>
        <div dangerouslySetInnerHTML={{ __html: this.classDetails.textbook }} />
      </React.Fragment>
    );
  }

  public getInstructorMarkup(): JSX.Element {
    let instructor = <><span>Instructor</span> TBD</>;
    if (this.classDetails.instructorName.trim().length !== 0) {
      const instructorName = this.classDetails.instructorName;
      const instructorProfileURL = this.searchURL + this.classDetails.profile;
      let userProfile = <>{instructorName}</>;
      if (this.classDetails.profile !== undefined && this.classDetails.profile.length !== 0) {
        userProfile = <a target="_blank" href={instructorProfileURL}>{instructorName}</a>;
      }
      instructor = (
        <><span>Instructor</span>{userProfile}</>
      );
    }
    return instructor;
  }

  public getClassFullTitle(): JSX.Element {
    const topicLowercase = this.classDetails.topic.toLowerCase();
    const topic = (topicLowercase.trim().length !== 0) ? `: ${Utils.toCapitalizeCase(topicLowercase)}` : '';
    const tooltip = this.getTooltipMarkup();
    return (
      <div className="course-name">
        <strong>
          {`${this.classDetails.subject} ${this.classDetails.catalogNo}`}
          {` - ${this.classDetails.title}${topic}`}
        </strong>
        <Popover
          content={this.classDetails.longDescription}
          position={'auto'}
          popoverClassName={Classes.POPOVER_CONTENT_SIZING}
        >
          {tooltip}
        </Popover>
      </div >
    );
  }

  public getCourseHeaderMarkup(): JSX.Element {
    const classType = ClassSearchUtils.getClassType(this.classDetails);
    const title = this.getClassFullTitle();
    return (
      <div className="course-header">
          {title}
          <div className="course-details">
              <span>Section {this.classDetails.classSection} • </span>
              <span>{classType} • </span>
              <span className="course-id">Class No. {this.classDetails.classNumber}</span>
          </div>
      </div>
    );
  }

  public isCurrentTerm(): boolean {
    return (this.props.currentTerm === this.classDetails.quarter) || parseInt(this.classDetails.quarter, 10) >= 2204;
  }

  public getClassStatusMarkup(): JSX.Element {
    let classStatus = 'Closed';
    let classStatusClassName = 'course-status course-status--closed';
    if (this.isCurrentTerm()) {
      classStatus = ClassSearchUtils.getClassStatus(this.classDetails);
      if (classStatus === 'Open') {
        classStatusClassName = 'course-status course-status--open';
      }
    }
    return (
      <React.Fragment>
        <div className={classStatusClassName}><span />{classStatus}</div>
      </React.Fragment>
    );
  }

  public getClassAvailabilityMarkup(): JSX.Element {
    const noOfSeatsAvailable = ClassSearchUtils.getNoOfAvailableSeats(this.classDetails);
    const waitlistMarkup = <div>{this.getWaitlistMarkup()}</div>;
    if (!this.isCurrentTerm()) {
      return (<React.Fragment/>);
    }
    if (ClassSearchUtils.getClassStatus(this.classDetails) === 'Closed') {
      // @Todo add markup to shows seats available
      return (<>{waitlistMarkup}</>);
    }
    if (ClassSearchUtils.isWaitlist(this.classDetails)) {
      return (<>{waitlistMarkup}</>);
    }
    return (
      <div className="course-availability">
      Seats Available: <span>{noOfSeatsAvailable}/{this.classDetails.enrolledCapacity}</span>
      </div>
    );
  }

  public getWaitlistMarkup(): JSX.Element {
    const waitlistNo = this.classDetails.waitlistTotal;
    if (!this.isCurrentTerm()) {
      return (<React.Fragment/>);
    }
    // @Todo delete this
    if (this.classDetails.waitlistCapacity === 0) {
      return (<div className="course-availability">No Waitlist</div>);
    }
    return (
      <div className="course-availability">
      Waitlist: <span>{waitlistNo}/{this.classDetails.waitlistCapacity}</span>
      </div>
    );
  }

  public getTooltipMarkup(): JSX.Element {
    return (
      <div className={Classes.TOOLTIP_INDICATOR}>
        <button className="course-info-btn">
          <i className="fas fa-info-circle" arial-label="hidden" />
          <span className="sr-only">{`${this.classDetails.subject} ${this.classDetails.catalogNo}`}</span>
        </button>
      </div>
    );
  }

  public isNonRegularSession(): boolean {
    const sessionCodes = [
      '10W',
      '6W1',
      '6W2',
      '3W1',
    ];
    return (sessionCodes.includes(this.classDetails.sessionCode.trim()));
  }

  public getSessionMarkup(): JSX.Element {
    if (this.isNonRegularSession()) {
      const session = ClassSearchUtils.getSessionCode(this.classDetails);
      return (<><li><span>Session </span>{session}</li></>);
    }
    return <></>;
  }

  public getRoomNumberMarkup(): JSX.Element {
    const roomNumber = ClassSearchUtils.getRoomNumber(this.classDetails);
    if (this.classDetails.instructionMode !== 'FO') {
      return (<li><span>Room </span>{roomNumber}</li>);
    }
    return <></>;
  }

  public getCourseMaterialIconMarkup(): JSX.Element {
    let zeroCostIcon = null;
    let lowCostIcon = null;
    const courseAttrArr = this.classDetails.courseAttr.split(', ').map(_ele => _ele.trim());
    if (courseAttrArr.includes('Zero Cost Course Materials')) {
      zeroCostIcon = (
        <img
          src="https://www.csusb.edu/sites/default/files/zero-cost-book-icon-big.png"
          alt="Zero Cost Course Materials"
        />
      );
    }
    if (courseAttrArr.includes('Low Cost Course Materials')) {
      lowCostIcon = (
        <img
          src="https://www.csusb.edu/sites/default/files/low-cost-book-icon-big.png"
          alt="Low Cost Course Materials"
        />
      );
    }
    if (!zeroCostIcon && !lowCostIcon) {
      return <></>;
    }
    return (<div className="course--icons">{zeroCostIcon}{lowCostIcon}</div>);
  }

}
