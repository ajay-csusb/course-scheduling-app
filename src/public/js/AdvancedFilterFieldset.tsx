import React from 'react';
import { Collapse, Label } from '@blueprintjs/core';
import { SelectListCourseAttr } from './SelectListCourseAttr';
import { SelectListSessionCode } from './SelectListSessionCode';
import { SelectListGeClassesAttributes } from './SelectListGeClassesAttributes';
import { IClassSearchFormProps } from './ClassSearchForm';

interface IAdvancedFilterFieldsetState {
  isOpen?: boolean;
}

export class AdvancedFilterFieldset extends React.Component<
    IClassSearchFormProps, IAdvancedFilterFieldsetState> {
  constructor(props: IClassSearchFormProps) {
    super(props);
    this.state = {
      isOpen: false,
    };
    this.handleClick = this.handleClick.bind(this);
  }

  public render() {
    const courseAttr = this.getSelectListCourseAttrComponent();
    const classNumber = this.getClassNumberComponent();
    const sessionCode = this.getSelectListSessionCodeComponent();
    const geClassesAttribute = this.getSelectListGeClassesAttributesComponent();
    const arrow = (this.state.isOpen) ? <i className="fal fa-chevron-up"/> : <i className="fal fa-chevron-down"/>;
    return (
      <React.Fragment>
        <a id="additional-filters" onClick={this.handleClick}>
          {this.state.isOpen ? 'Hide' : ''} Additional Search Criteria {arrow}
        </a>
        <Collapse isOpen={this.state.isOpen} >
          <div className="row">
            <div className="col-6 col-md-4">
              <div className="form-item">{geClassesAttribute}</div>
            </div>
            <div className="col-6 col-md-4">
              <div className="form-item">{courseAttr}</div>
            </div>
            <div className="col-6 col-md-4">
              <div className="form-item">{classNumber}</div>
            </div>
            <div className="col-6 col-md-4">
              <div className="form-item">{sessionCode}</div>
            </div>
          </div>
        </Collapse>
      </React.Fragment>
    );
  }

  private handleClick = () => {
    this.setState({ isOpen: !this.state.isOpen });
  }

  private getSelectListCourseAttrComponent(): JSX.Element {
    return (
      <SelectListCourseAttr
        courseAttr={this.props.courseAttr}
        onChangeOfCourseAttr={this.props.onChangeOfCourseAttr}
      />
    );
  }

  private getClassNumberComponent(): JSX.Element {
    return (
      <React.Fragment>
        <Label htmlFor="class-number">Class Number</Label>
        <input
          id="class-number"
          className="class-number"
          type="text"
          placeholder="E.g. 60256"
          dir="auto"
          onChange={this.props.onChangeOfClassNo}
          value={this.props.classNo}
          onKeyDown={this.props.onKeyDown}
        />
      </React.Fragment>
    );
  }

  private getSelectListSessionCodeComponent(): JSX.Element {
    return (
      <SelectListSessionCode
        sessionCode={this.props.sessionCode}
        onChangeOfSessionCode={this.props.onChangeOfSessionCode}
      />
    );
  }

  private getSelectListGeClassesAttributesComponent(): JSX.Element {
    return (
      <SelectListGeClassesAttributes
        geClassesAttributes={this.props.geClassesAttributes}
        geClassesAttribute={this.props.geClassesAttribute}
        onChangeOfGeClassesAttribute={this.props.onChangeOfGeClassesAttribute}
      />
    );
  }

}
