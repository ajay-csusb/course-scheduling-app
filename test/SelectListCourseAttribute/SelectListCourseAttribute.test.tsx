import { mount } from 'enzyme';
import * as React from 'react';
import { SelectListCourseAttr } from '../../src/public/js/SelectListCourseAttr';
// tslint:disable:max-line-length
describe('<SelectListCourseAttr>', () => {

  describe('Course attribute select list', () => {
    test('snapshot', () => {
      const courseAttributeSelectWrapper = mount(<SelectListCourseAttr courseAttr='all' onChangeOfCourseAttr={() => {}}/>);
      expect(courseAttributeSelectWrapper).toMatchSnapshot();
    });
  });
});
