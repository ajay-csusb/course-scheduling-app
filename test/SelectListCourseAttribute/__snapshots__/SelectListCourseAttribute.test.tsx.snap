// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectListCourseAttr> Given a class search form snapshot 1`] = `
Array [
  <HTMLSelect
    className="course-attribute"
    id="course-attribute"
    onChange={[Function]}
    options={
      Array [
        Object {
          "label": "All",
          "value": "all",
        },
        Object {
          "label": "Asian Studies",
          "value": "ASTD",
        },
        Object {
          "label": "Chicano(a)/Latino(a) Studies",
          "value": "CLST",
        },
        Object {
          "label": "eBook",
          "value": "EBK",
        },
        Object {
          "label": "Ethnic Studies",
          "value": "ETHN",
        },
        Object {
          "label": "GE Designation",
          "value": "DES",
        },
        Object {
          "label": "Gender and Sexuality Studies",
          "value": "GSS",
        },
        Object {
          "label": "Latin American Studies",
          "value": "LTAM",
        },
        Object {
          "label": "Low Cost Course Materials",
          "value": "LCCM",
        },
        Object {
          "label": "Service Learning",
          "value": "CSLI",
        },
        Object {
          "label": "Study Abroad",
          "value": "SA",
        },
        Object {
          "label": "Women's Studies",
          "value": "WSTD",
        },
        Object {
          "label": "Zero Cost Course Materials",
          "value": "ZCCM",
        },
      ]
    }
    value="all"
  >
    <div
      className="bp3-html-select course-attribute"
    >
      <select
        id="course-attribute"
        multiple={false}
        onChange={[Function]}
        value="all"
      >
        <option
          key="all"
          label="All"
          value="all"
        >
          All
        </option>
        <option
          key="ASTD"
          label="Asian Studies"
          value="ASTD"
        >
          Asian Studies
        </option>
        <option
          key="CLST"
          label="Chicano(a)/Latino(a) Studies"
          value="CLST"
        >
          Chicano(a)/Latino(a) Studies
        </option>
        <option
          key="EBK"
          label="eBook"
          value="EBK"
        >
          eBook
        </option>
        <option
          key="ETHN"
          label="Ethnic Studies"
          value="ETHN"
        >
          Ethnic Studies
        </option>
        <option
          key="DES"
          label="GE Designation"
          value="DES"
        >
          GE Designation
        </option>
        <option
          key="GSS"
          label="Gender and Sexuality Studies"
          value="GSS"
        >
          Gender and Sexuality Studies
        </option>
        <option
          key="LTAM"
          label="Latin American Studies"
          value="LTAM"
        >
          Latin American Studies
        </option>
        <option
          key="LCCM"
          label="Low Cost Course Materials"
          value="LCCM"
        >
          Low Cost Course Materials
        </option>
        <option
          key="CSLI"
          label="Service Learning"
          value="CSLI"
        >
          Service Learning
        </option>
        <option
          key="SA"
          label="Study Abroad"
          value="SA"
        >
          Study Abroad
        </option>
        <option
          key="WSTD"
          label="Women's Studies"
          value="WSTD"
        >
          Women's Studies
        </option>
        <option
          key="ZCCM"
          label="Zero Cost Course Materials"
          value="ZCCM"
        >
          Zero Cost Course Materials
        </option>
      </select>
      <Blueprint3.Icon
        icon="double-caret-vertical"
      >
        <span
          className="bp3-icon bp3-icon-double-caret-vertical"
          icon="double-caret-vertical"
        >
          <svg
            data-icon="double-caret-vertical"
            height={16}
            viewBox="0 0 16 16"
            width={16}
          >
            <desc>
              double-caret-vertical
            </desc>
            <path
              d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
              fillRule="evenodd"
              key="0"
            />
          </svg>
        </span>
      </Blueprint3.Icon>
    </div>
  </HTMLSelect>,
  <div
    className="bp3-html-select course-attribute"
  >
    <select
      id="course-attribute"
      multiple={false}
      onChange={[Function]}
      value="all"
    >
      <option
        key="all"
        label="All"
        value="all"
      >
        All
      </option>
      <option
        key="ASTD"
        label="Asian Studies"
        value="ASTD"
      >
        Asian Studies
      </option>
      <option
        key="CLST"
        label="Chicano(a)/Latino(a) Studies"
        value="CLST"
      >
        Chicano(a)/Latino(a) Studies
      </option>
      <option
        key="EBK"
        label="eBook"
        value="EBK"
      >
        eBook
      </option>
      <option
        key="ETHN"
        label="Ethnic Studies"
        value="ETHN"
      >
        Ethnic Studies
      </option>
      <option
        key="DES"
        label="GE Designation"
        value="DES"
      >
        GE Designation
      </option>
      <option
        key="GSS"
        label="Gender and Sexuality Studies"
        value="GSS"
      >
        Gender and Sexuality Studies
      </option>
      <option
        key="LTAM"
        label="Latin American Studies"
        value="LTAM"
      >
        Latin American Studies
      </option>
      <option
        key="LCCM"
        label="Low Cost Course Materials"
        value="LCCM"
      >
        Low Cost Course Materials
      </option>
      <option
        key="CSLI"
        label="Service Learning"
        value="CSLI"
      >
        Service Learning
      </option>
      <option
        key="SA"
        label="Study Abroad"
        value="SA"
      >
        Study Abroad
      </option>
      <option
        key="WSTD"
        label="Women's Studies"
        value="WSTD"
      >
        Women's Studies
      </option>
      <option
        key="ZCCM"
        label="Zero Cost Course Materials"
        value="ZCCM"
      >
        Zero Cost Course Materials
      </option>
    </select>
    <Blueprint3.Icon
      icon="double-caret-vertical"
    >
      <span
        className="bp3-icon bp3-icon-double-caret-vertical"
        icon="double-caret-vertical"
      >
        <svg
          data-icon="double-caret-vertical"
          height={16}
          viewBox="0 0 16 16"
          width={16}
        >
          <desc>
            double-caret-vertical
          </desc>
          <path
            d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
            fillRule="evenodd"
            key="0"
          />
        </svg>
      </span>
    </Blueprint3.Icon>
  </div>,
]
`;
